lib/src/front_end/ast_node_visitor.dart

```diff
@@ -1310,7 +1405,30 @@ class AstNodeVisitor extends ThrowingAstVisitor<Piece> with PieceFactory {
 
   @override
   Piece visitSuperFormalParameter(SuperFormalParameter node) {
-    throw UnimplementedError();
+    if (node.parameters case var parameters?) {
+      // A function-typed super parameter like:
+      //
+      // ```
+      // C(super.fn(parameter));
+      // ```
+      return createFunctionType(
```

Comment:
Yeah, this is a fair point. I started using backticks during the rewrite because I thought it was maybe a little easier to write than using indentation. But in retrospect the backticks require an extra pair of lines which is lame.

How about I fix all of them in a separate PR?